From 37a1be259139b0ce6b73481409f720260fc3cbcf Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 4 Nov 2021 09:29:23 +0200
Subject: [PATCH 06/24] dts/s32g-evb: Enable FTM0 & FTM1

commit 37a1be259139b0ce6b73481409f720260fc3cbcf from
https://source.codeaurora.org/external/autobsps32/linux

This enables FTM0 & FTM1 modules using the following
channels:
	* FTM0, channels: 2, 3, 4
	* FTM1, channels: 1, 2, 3, 4

Issue: ALB-8070
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../boot/dts/freescale/fsl-s32gxxxa-evb.dtsi  | 48 +++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32gxxxa-evb.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32gxxxa-evb.dtsi
index 81a43e6f7b9c..d88ab61fc84e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32gxxxa-evb.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32gxxxa-evb.dtsi
@@ -447,6 +447,21 @@
 			>;
 		};
 
+		pinctrl0_ftm0: ftm0grp {
+			fsl,pins = <
+				S32_GEN1_PAD_PB_02__FTM0_CH2_O
+
+			>;
+		};
+
+		pinctrl0_ftm1: ftm1grp {
+			fsl,pins = <
+				S32_GEN1_PAD_PB_13__FTM1_CH2_O
+				S32_GEN1_PAD_PB_15__FTM1_CH3_O
+				S32_GEN1_PAD_PC_00__FTM1_CH4_O
+			>;
+		};
+
 		pinctrl_gpio0: gpiogrp0 {
 			fsl,pins = <
 				S32_GEN1_PAD_PA0__SIUL_GPIO0
@@ -691,6 +706,26 @@
 			>;
 		};
 
+		pinctrl1_ftm0: ftm0grp {
+			fsl,pins = <
+				S32_GEN1_PAD_PB_02__FTM0_CH2_I
+				S32_GEN1_PAD_PL_01__FTM0_CH1_O
+				S32_GEN1_PAD_PL_01__FTM0_CH1_I
+				S32_GEN1_PAD_PL_04__FTM0_CH3_O
+				S32_GEN1_PAD_PL_04__FTM0_CH3_I
+				S32_GEN1_PAD_PL_05__FTM0_CH4_O
+				S32_GEN1_PAD_PL_05__FTM0_CH4_I
+			>;
+		};
+
+		pinctrl1_ftm1: ftm1grp {
+			fsl,pins = <
+				S32_GEN1_PAD_PB_13__FTM1_CH2_I
+				S32_GEN1_PAD_PB_14__FTM1_CH3_I
+				S32_GEN1_PAD_PC_00__FTM1_CH4_I
+			>;
+		};
+
 		pinctrl_gpio1: gpiogrp1 {
 			fsl,pins = <
 				S32_GEN1_PAD_PH0__SIUL_GPIO112
@@ -942,3 +977,16 @@
 &llce_can15 {
 	status = "okay";
 };
+
+&pwm0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_ftm0>, <&pinctrl1_ftm0>;
+	status = "okay";
+};
+
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_ftm1>, <&pinctrl1_ftm1>;
+	status = "okay";
+};
+
-- 
2.17.1

